discord-bot/src/commands/market-react-bets.ts

```diff
@@ -0,0 +1,147 @@
+import { FullMarket } from 'common/api-market-types'
+import {
+  AttachmentBuilder,
+  ChatInputCommandInteraction,
+  EmbedBuilder,
+  MessageReaction,
+  SlashCommandBuilder,
+  TextChannel,
+  User,
+} from 'discord.js'
+import { customEmojis, emojis, getBettingEmojisAsStrings } from '../emojis.js'
+import {
+  getCurrentMarketDescription,
+  getOpenBinaryMarketFromSlug,
+  getSlug,
+  handleReaction,
+} from '../helpers.js'
+import { messagesHandledViaInteraction } from '../storage.js'
+
+export const data = new SlashCommandBuilder()
+  .setName('market')
+  .setDescription('Link to a market that other users can bet on with reactions')
+  .addStringOption((option) =>
+    option
+      .setName('link')
+      .setDescription('The link to the market to bet on')
+      .setRequired(true)
+  )
+
+export async function execute(interaction: ChatInputCommandInteraction) {
+  const link = interaction.options.getString('link')
+  if (!link || !link.startsWith('https://manifold.markets/')) {
+    await interaction.reply(
+      'You must specify a market link starting with https://manifold.markets/'
+    )
+    return
+  }
+  const slug = getSlug(link)
+  if (!slug) {
+    await interaction.reply(
+      'Invalid market link, could not find slug from link'
+    )
+    return
+  }
+  const market = await getOpenBinaryMarketFromSlug(slug, (error) =>
+    interaction.reply(error)
```

Comment:
The error callback and the fact that we return without awaiting this are pretty weird.